From: Thomas Renninger <trenn@suse.de>
Subject: Use acpi_device_id for IA64 HP driver, otherwise those fail to boot
Patch-mainline: yes
References: none


commit 7091138fb762aed22317b4ff91eb211e7da3865c

    [IA64] Use new acpi_device_id struct for HID init in ia64/hp/common/sba_iommu.c
    
    Forgot to adjust this one with the acpi autoloading patches
    in commit 8c8eb78f673c07b60f31751e1e47ac367c60c6b7
    
    Acked-by: Myron Stowe <myron.stowe@hp.com>
    Acked-by: Len Brown <len.brown@intel.com>
    Signed-off-by: Thomas Renninger <trenn@suse.de>
    Signed-off-by: Tony Luck <tony.luck@intel.com>

diff --git a/arch/ia64/hp/common/sba_iommu.c b/arch/ia64/hp/common/sba_iommu.c
index cd4adf5..e980e7a 100644
--- a/arch/ia64/hp/common/sba_iommu.c
+++ b/arch/ia64/hp/common/sba_iommu.c
@@ -2015,9 +2015,14 @@ acpi_sba_ioc_add(struct acpi_device *device)
 	return 0;
 }
 
+static const struct acpi_device_id hp_ioc_iommu_device_ids[] = {
+	{"HWP0001", 0},
+	{"HWP0004", 0},
+	{"", 0},
+};
 static struct acpi_driver acpi_sba_ioc_driver = {
 	.name		= "IOC IOMMU Driver",
-	.ids		= "HWP0001,HWP0004",
+	.ids		= hp_ioc_iommu_device_ids,
 	.ops		= {
 		.add	= acpi_sba_ioc_add,
 	},
